Extension/src/LanguageServer/client.ts

```diff
@@ -483,24 +491,53 @@ class DefaultClient implements Client {
     }
 
     /*************************************************************************************
-     * wait until the language client is ready for use before attempting to send messages
+     * wait until the all pendingTasks are complete (e.g. language client is ready for use)
+     * before attempting to send messages
      *************************************************************************************/
+    public runBlockingTask(task: Thenable<any>): Thenable<any> {
```

Comment:
>any [](start = 42, length = 3)

Should this function be generic (i.e. accept type T), so this can be Thenable<T>?

Reply:
Same for requestWhenReady below

---
In reply to: [185363898](https://github.com/Microsoft/vscode-cpptools/pull/1804#discussion_r185363898) [](ancestors = 185363898)